.test-infra/jenkins/DockerPublisher.groovy

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonJobProperties as common
+
+class DockerPublisher {
+  private String repositoryRoot
+
+  DockerPublisher(String repositoryRoot = 'gcr.io/apache-beam-testing/beam_portability') {
```

Comment:
This class is all about docker and abstracts away from other aspects (portability, flink). I like that but in that case we can't have this default value (`'gcr.io/apache-beam-testing/beam_portability'`) because it mentions portability. Maybe it should be defined in SDKHarnessPublisher?